lib/bank_routing/routing_numbers.rb

```diff
@@ -108,18 +108,17 @@ def options
 		def get_raw_data
 			if options[:fetch_fed_data]
 				log.info "Getting new bank routing data from: #{options[:routing_data_url]}"
-				require 'typhoeus'
-				response = Typhoeus::Request.post(options[:routing_data_agreement_url], body: 'agreementValue=Agree')
-				set_cookie = response.headers["Set-Cookie"].match(/JSESSIONID=.*?;/).to_s
-				Typhoeus::Request.get(options[:routing_data_url], ssl_verifypeer: false,
-					headers: { 'Cookie' => set_cookie } ).body
+				require 'bank_routing/http_adapter'
+				HttpAdapter.new(options).fetch_routing_directory
 			else
 				log.info "Using routing data from local file at: #{options[:routing_data_file]}"
 				File.new(options[:routing_data_file])
 			end
 		end
 
 		def load_routing_numbers(data=get_raw_data)
+			return unless get_raw_data&.respond_to?(:each_line)
```

Comment:
maybe refer to data instead of get_raw_data here, or remove data=get_raw_data above?

Reply:
yikes, yeah should be using the data var